bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorImpl.java

```diff
@@ -49,20 +47,28 @@
  * @author Chris Jackson - Handle checks on multiple selection parameters
  */
 @Component
+@NonNullByDefault
 public final class ConfigDescriptionValidatorImpl implements ConfigDescriptionValidator {
 
-    private final Logger logger = LoggerFactory.getLogger(ConfigDescriptionValidatorImpl.class);
+    private static final List<ConfigDescriptionParameterValidator> VALIDATORS = List.of( //
+            ConfigDescriptionParameterValidatorFactory.createRequiredValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createTypeValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createMinMaxValidator(), //
+            ConfigDescriptionParameterValidatorFactory.createPatternValidator() //
+    );
 
-    private ConfigDescriptionRegistry configDescriptionRegistry;
-    private TranslationProvider translationProvider;
+    private final Logger logger = LoggerFactory.getLogger(ConfigDescriptionValidatorImpl.class);
 
-    private static final List<ConfigDescriptionParameterValidator> VALIDATORS = Collections
-            .unmodifiableList(Arrays.asList(ConfigDescriptionParameterValidatorFactory.createRequiredValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createTypeValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createMinMaxValidator(),
-                    ConfigDescriptionParameterValidatorFactory.createPatternValidator()));
+    private final ConfigDescriptionRegistry configDescriptionRegistry;
+    private final TranslationProvider translationProvider;
+    private @NonNullByDefault({}) BundleContext bundleContext;
 
-    private BundleContext bundleContext;
+    @Activate
+    public ConfigDescriptionValidatorImpl(final @Reference ConfigDescriptionRegistry configDescriptionRegistry,
+            final @Reference TranslationProvider translationProvider) {
```

Comment:
This makes the `TranslationProvider` and the `ConfigDescriptionRegistry` mandatory, they were optional before. That change makes sense to me, but did you check if this does not create circular dependencies?

Reply:
No, but afair the `TranslationProvider` does not reference any other services. The `ConfigDescriptionRegistry` only references optional provider.